### PR TITLE
Updated validity dependencies

### PR DESCRIPTION
**Pull request by CSVdB**

### CODE REVIEW

**Comment by CSVdB** [comment]
This is a version bump.

File: zifter/src/Zifter/Zift/Types.hs [hunk 1]
```diff
@@ -108,8 +106,8 @@ data ZiftResult a
     deriving (Show, Eq, Generic)
 
 instance Validity a => Validity (ZiftResult a) where
-    isValid (ZiftSuccess a) = isValid a
-    isValid _ = True
+    validate (ZiftSuccess a) = validate a
```
Comment (NorfairKing) [inline zifter/src/Zifter/Zift/Types.hs]: This is not semantics preserving. You'll have to take a very close look at the versioun bounds on validity.

File: zifter/test/Zifter/Zift/Gen.hs [hunk 2]
```diff
@@ -53,15 +53,13 @@ instance (Validity a) => Validity (RGB a) where
     validate RGB {..} =
         (channelRed <?!> "channelRed") <> (channelGreen <?!> "channelGreen") <>
         (channelBlue <?!> "channelBlue")
-    isValid = isValidByValidating
```
Comment (NorfairKing) [inline zifter/test/Zifter/Zift/Gen.hs]: Same comment.

File: zifter/test/Zifter/Zift/Gen.hs [hunk 3]
```diff
@@ -53,15 +53,13 @@ instance (Validity a) => Validity (RGB a) where
...
 
 instance (Ord a, Floating a, GenUnchecked a) => GenUnchecked (Colour a) where
     genUnchecked = sRGB24 <$> genUnchecked <*> genUnchecked <*> genUnchecked
     shrinkUnchecked _ = []
 
 instance (Floating a, RealFrac a, Validity a) => Validity (Colour a) where
-    isValid = isValid . toSRGB24
-    validate = validateByChecking "Colour"
+    validate x = isValid (toSRGB24 x) <?@> "Colour"
```
Comment (NorfairKing) [inline zifter/test/Zifter/Zift/Gen.hs]: same comment.

File: zifter/package.yaml [hunk 4]
```diff
@@ -52,9 +52,9 @@ tests:
     - zifter
     - QuickCheck >=2.9 && <2.11
     - colour >=2.3 && <2.4
-    - genvalidity >=0.3 && <0.5
-    - genvalidity-hspec >=0.3 && <0.6
-    - genvalidity-path >=0.1 && <0.3
+    - genvalidity >=0.4 && <0.5
+    - genvalidity-hspec >=0.5 && <0.6
+    - genvalidity-path >=0.2 && <0.3
```
Comment (NorfairKing) [inline zifter/package.yaml]: It's better to upgrade the upper bound if necessary, but maintain backward compatibility.

File: stack.yaml [hunk 5]
```diff
@@ -8,4 +8,7 @@ packages:
 - zifter-hindent
 - zifter-hlint
 - zifter-stack
+extra-deps:
+- validity-0.4.0.4
+- genvalidity-0.4.0.4
```
Comment (NorfairKing) [inline stack.yaml]: validity is on stackage. Better to wait until the right version is available there.

File: lts-8.yaml [hunk 6]
```diff
@@ -10,10 +10,10 @@ packages:
 - zifter-stack
 
 extra-deps:
-- genvalidity-0.4.0.0
+- genvalidity-0.4.0.4
 - genvalidity-hspec-0.5.0.0
 - genvalidity-path-0.2.0.0
 - genvalidity-property-0.1.0.0
-- validity-0.4.0.0
+- validity-0.4.0.4
```
Comment (NorfairKing) [inline lts-8.yaml]: This will probably get us into problems because validity wasn't available yet when this lts was made.

**Review by NorfairKing — CHANGES_REQUESTED** [review]

**Pull request closed without merging**